pkg/forman/forman.go

```diff
@@ -0,0 +1,99 @@
+package forman
+
+import (
+	"time"
+
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/types"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/utils"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/workers"
+
+	"go.uber.org/zap"
+)
+
+type provider struct {
+	working    bool
+	next       *time.Time
+	alarmClock *time.Timer
+}
+
+func GoForman(
+	logger *zap.Logger,
+	workerCount int,
+	reconcile func(*zap.Logger, types.Request) types.Result,
+) chan types.Request {
+	requests := make(chan types.Request)
+
+	go func() {
+		toWorkers := make(chan types.Request)
+		results := make(chan utils.Pair[types.Request, types.Result])
+		providers := map[string]*provider{}
+
+		sendToWorkers := func(req types.Request) {
+			go func() { toWorkers <- req }()
+		}
+
+		logger.Info("Starting", zap.Int("workers", workerCount))
+		for i := 0; i < workerCount; i++ {
+			workers.GoWorker(toWorkers, results, func(req types.Request) types.Result {
+				return reconcile(logger, req)
+			})
```

Comment:
No need for that, we can enhance the worker API to include a logger as well. Mainly as it is a standard

Reply:
Sorry didnt get the suggestion